### ISSUE TITLE
Expose a simpler `pathWith`

### ISSUE THREAD

**Issue by wavebeem** [ramda/ramda#189]
There are currently a very large chunk of `path`-related functions in Ramda. `path` makes sense as the default use case. `pathOn` makes sense though is just sugar over `pathWith` really. The way I see `path` is two cases:
1. By default, use dots as the separators in a path string
2. If you need more customizability, just have `pathList :: [String] -> Object -> Any`

`pathWith` seems far too complicated:

``` haskell
pathWith :: (String -> [String]) -> String -> Object -> Any
```

In short, my suggestion is axe `pathWith` and expose the internal `path` in a curried form as `pathList` or something.
